src/isomorphic/classic/element/ReactElement.js

```diff
@@ -231,13 +223,30 @@ ReactElement.createElement = function(type, config, children) {
   // Children can be more than one argument, and those are transferred onto
   // the newly allocated props object.
   var childrenLength = arguments.length - 2;
+  var childArray;
   if (childrenLength === 1) {
-    props.children = children;
+    if (Array.isArray(children)) {
+      childArray = children.slice(0);
```

Comment:
Yes, that's unfortunate. :( Do you think there's a better way to do this? 

However, it should be more performant than the current approach, which slices in the `ReactElement()` function (https://github.com/facebook/react/blob/master/src/isomorphic/classic/element/ReactElement.js#L138) for every component that has more than one child. The new one slices in `createElement` and only when an user explicitly passes an array as `children`. If this makes sense then I guess the question is whether we want to slice in prod.